tests/unit/parallel/executors/minimal_timed_async_executor_v1.cpp

```diff
@@ -0,0 +1,196 @@
+//  Copyright (c) 2007-2016 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/hpx.hpp>
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <boost/range/functions.hpp>
+
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <functional>
+#include <string>
+#include <utility>
+#include <type_traits>
+#include <vector>
+
+using namespace std::chrono;
+
+///////////////////////////////////////////////////////////////////////////////
+hpx::thread::id sync_test(int passed_through)
+{
+    HPX_TEST_EQ(passed_through, 42);
+    return hpx::this_thread::get_id();
+}
+
+void apply_test(hpx::lcos::local::latch& l, hpx::thread::id& id,
+    int passed_through)
+{
+    HPX_TEST_EQ(passed_through, 42);
+    id = hpx::this_thread::get_id();
+    l.count_down(1);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename Executor>
+void test_timed_apply(Executor& exec)
+{
+    {
+        hpx::lcos::local::latch l(3);
+        hpx::thread::id id;
+
+        typedef hpx::parallel::timed_executor_traits<Executor> traits;
+
+        using hpx::util::placeholders::_1;
+        traits::apply_execute_after(exec, milliseconds(10),
+            hpx::util::bind(&apply_test, std::ref(l), std::ref(id), _1), 42
+        );
+
+        traits::apply_execute_after(exec, milliseconds(10),
+            &apply_test, std::ref(l), std::ref(id), 42);
+
+        l.count_down_and_wait();
+
+        HPX_TEST(id != hpx::this_thread::get_id());
+    }
+
+    {
+        hpx::lcos::local::latch l(3);
+        hpx::thread::id id;
+
+        typedef hpx::parallel::timed_executor_traits<Executor> traits;
+        traits::apply_execute_at(exec, steady_clock::now() + milliseconds(10),
+            hpx::util::deferred_call(
+                &apply_test, std::ref(l), std::ref(id), 42));
+
+        traits::apply_execute_at(exec, steady_clock::now() + milliseconds(10),
+            &apply_test, std::ref(l), std::ref(id), 42);
```

Comment:
No, this test is for the old interface (hence the `v1`). It will be removed once we completely deprecate it.